ball_detect/main.c

```diff
@@ -39,88 +44,61 @@ void setup(void) {
 	DDRD |= 1 << 3;
 }
 
+#define waitForTX() while (!(UCSR0A & 1<<UDRE0))
+
+void sendc(uint8_t ch) {
+  waitForTX();
+  UDR0 = ch;
+}
+
+
+
 ISR(USART_RX_vect) {
 	
 }
 
 ISR(PCINT0_vect) {
-	static uint8_t portb_history = 0xFF;
-	uint8_t changed_bits;
-	uint8_t pinstate;
-
 	pinstate = PINB;
 	PORTD ^= 1 << 3;
 	changed_bits = pinstate ^ portb_history;
 	portb_history = pinstate;  
-	int ct;
+	int ct = 0;
 	for (ct = 0;ct < 4;ct++) {
-		if (changed_bits & (1 << ct)) {
-       		if (pinstate & (1 << ct)) {
-				helper[ct] = TCNT1H; 
-		   		if(counters[ct] < 10) { 
-					counters[ct]++; 
-				} else {
-	        		counters[ct] = 0;
-					pulses[ct] = pulses[ct]/10;
-   	     		}  			    	 	
+		if (changed_bits & (1 << ct)) { 
+       		if (pinstate & (1 << ct)) { 
+				helper[ct] = TCNT1; 
 			} else {
-				pulse = TCNT1;
-				if(pulse < helper[ct]) {
-					pulses[ct] += pulse+2000000-helper[ct];
-					helper[ct] = 0;
-				} else {
-					pulses[ct] += pulse - helper[ct];
-				}
+				pulse[ct] = TCNT1;
+				pulses[ct] = abs(pulse[ct]-helper[ct]);
 			}
 		}
-	}
-    
+    }
 	if(changed_bits & (1 << ULTRASONIC_SENSOR)) {
 		
 	}
 
 }
 
-#define waitForTX() while (!(UCSR0A & 1<<UDRE0))
-
-void sendc(uint8_t ch) {
-  waitForTX();
-  UDR0 = ch;
-}
-
 int main(void) {
-	uint8_t i;
+	uint8_t j, index = 0;
+ 	uint32_t  small;
```

Comment:
I think only one space should be between type specifier and variable name, not 2.